src/realm/alloc_slab.cpp

```diff
@@ -380,7 +381,7 @@ MemRef SlabAlloc::do_alloc(const size_t size)
 #ifdef REALM_SLAB_ALLOC_DEBUG
     malloc_debug_map[ref] = malloc(1);
 #endif
-
+    REALM_ASSERT_DEBUG(ref >= m_baseline);
     return MemRef(slab.addr, ref, *this);
 }
```

Comment:
I would argue that we should use `REALM_ASSERT()` in cases like this, and that `REALM_ASSERT_DEBUG()` should almost never be used.

My main point here is that `REALM_ASSERT()` is already an assertion that is only meant to be present in debug mode, and we must continue to think about it that way.